src/string_immutable.zig

```diff
@@ -144,16 +144,26 @@ pub inline fn isNPMPackageName(target: string) bool {
         '@' => true,
         else => return false,
     };
+
     var slash_index: usize = 0;
     for (target[1..], 0..) |c, i| {
         switch (c) {
             // Old packages may have capital letters
-            'A'...'Z', 'a'...'z', '0'...'9', '$', '-', '_', '.' => {},
+            'A'...'Z', 'a'...'z', '0'...'9', '-', '_', '.' => {},
+            '$' => {
```

Comment:
is `$` valid? It should be escaped by `encodeURIComponent` so we can remove the case for it

Reply:
🤯 sorry, fixed.